architecture/image_registry.adoc

```diff
@@ -10,4 +10,21 @@
 toc::[]
 
 == Overview
-OpenShift utilizes any server implementing the Docker registry API as a source of images, including the canonical DockerHub, private registries run by 3rd parties, and self hosted registries that implement the registry API
\ No newline at end of file
+
+OpenShift utilizes any server implementing the Docker registry API as a source of images, including the canonical Docker Hub, private registries run by 3rd parties, and the integrated OpenShift registry.
+
+=== Integrated OpenShift Registry
+
+OpenShift provides an integrated Docker registry that adds the ability to provision new image repositories on the fly (NOTE: this isn't 100% implemented yet), so users automatically have a place for their builds to push the resulting images.
+
+Whenever a new image is pushed to the integrated registry, the registry notifies OpenShift about the new image, passing along all the information about it (namespace, name, image metadata). Different pieces of OpenShift react to new images, creating new builds and deployments (as described elsewhere).
+
+=== 3rd Party Registries
+
+OpenShift can create containers using images from 3rd party registries, but these registries likely won't offer the same image notification support that the integrated OpenShift registry does. If not, OpenShift is able to poll the other registries for changes to image repositories. When new images are detected, the same build/deployment/etc reactions described above occur. (NOTE: polling is not implemented yet.)
+
+==== Authentication
+
+OpenShift is able to communicate with registries to access private image repositories using credentials supplied by the user. This allows OpenShift to push and pull images to/from private repositories.
+
+NOTE: This isn't implemented yet.
```

Comment:
This line will actually create a NOTE block, in contrast with your use of NOTE in lines 18 and 24 where it is just going to be rendered as text. Please be consistent; in this case I'd suggest taking this sentence and throwing it in parenthesis at the end of the paragraph in line 28 (the same way you do in line 24). That will prevent the fancy rendering.